### PR TITLE
Concept entry dart loops. Adresses issue #4249

### PR DESCRIPTION
**Pull request by kamlesh-ops**
### Description

- Added sections for different dart loops
- Closes issue #4249 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by kamlesh-ops** [comment]
Hello there.
I was unaware that this issue was assigned and I got busy in other tasks. I deeply regret the inconvenience.
I have made the pr. Please give it a review and ping me if there are any issues.

File: content/dart/concepts/loops/loops.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+1
+2
+3
+4
+5
+```
+
+## forEach Loop
+
+The for-each loop iterates over all elements in some container/collectible and passes the elements to some specific function.
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:88]:
```suggestion
The forEach loop iterates over all elements in a container or collection and invokes a specified function for each element.
```

File: content/dart/concepts/loops/loops.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+#### Syntax
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:90]:
```suggestion
### Syntax
```

File: content/dart/concepts/loops/loops.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+```pseudo
+collection.forEach(void f(value))
+```
+
+f(value) is a parameter which is used to call the function "f" for each value in the collection.
+
+#### Example
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:98]:
```suggestion
### Example
```

File: content/dart/concepts/loops/loops.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+1
+2
+3
+4
+5
+```
+
+## while Loop
+
+The while loop executes a block of code until the given condition is false. It is more beneficial when we don't know the number of executions.
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:120]: Using pronouns is not allowed. Remove 'we' and refine the definition.

File: content/dart/concepts/loops/loops.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+#### Syntax
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:122]:
```suggestion
### Syntax
```

File: content/dart/concepts/loops/loops.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+The output of the above code is,
+
+```shell
+Hello there
+Hello there
+Hello there
+Hello there
+```
+
+## do...while Loop
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:153]:
```suggestion
## do-while Loop
```

File: content/dart/concepts/loops/loops.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+The do…while loop is similar to the while loop but only difference is that it executes the loop statement and then check the given condition. Hence the block of code inside the loop gets executed atleast once for do...while loop.
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:155]:
```suggestion
The do-while loop is similar to the `while` loop, with the key difference being that it executes the loop statement before checking the given condition. Therefore, the block of code inside the loop is guaranteed to execute at least once
```

File: content/dart/concepts/loops/loops.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+#### Syntax
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:157]:
```suggestion
### Syntax
```

File: content/dart/concepts/loops/loops.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+
+```pseudo
+do {  
+    // loop body  
+} while(condition);  
+```
+
+#### Example
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:165]:
```suggestion
### Example
```

File: content/dart/concepts/loops/loops.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+f(value) is a parameter which is used to call the function "f" for each value in the collection.
+
+#### Example
+
+```dart
+void main()
+{
+  var nums = [1,2,3,4,5];
+  nums.forEach((var num)=> print(num));
```
Comment (letsbuildanything) [inline content/dart/concepts/loops/loops.md:104]: Using parameter name 'num' raise an issue. rename it!

**Review by letsbuildanything — CHANGES_REQUESTED** [review]
@kamlesh-ops I've listed below some changes. Please make them ASAP.
Also, When referring to elements as they appear in code (properties, methods, functions, commands, expressions, etc.), use backticks i.e. `for` instead of double quotes("for").

**Comment by letsbuildanything** [comment]
@kamlesh-ops It's been more than a month since the request was made. Please proceed with implementing all the suggested modifications.
Thank you!

**Comment by kamlesh-ops** [comment]
@letsbuildanything I am extremely sorry for the delay. I have made the changes. Please review. 
Thank you!

**Review by letsbuildanything — APPROVED** [review]
@kamlesh-ops Thank you for making the changes. It seems ready for the next review!

**Review by mohitO12-dev — APPROVED** [review]
Approved

**Pull request merged**